TF_Tooling_Web/src/Connection/tests/MessageReceiver.test.ts

```diff
@@ -0,0 +1,331 @@
+            originalRequest: message,
+            status: 'Success',
+            message: 'Successful Test',
+            ...content,
+        };
+        serviceConnection?.OnMessage(
+            new MessageEvent('message', {
+                data: JSON.stringify({
+                    action: actionCode,
+                    content: messageContent,
+                }),
+            })
+        );
+        return messageContent;
+    };
+
+    const mockOpen = () => serviceConnection?.webSocket.dispatchEvent(new Event('open'));
+
+    const mockHandshake = (consoleProperty: string) => {
+        const testFn = jest.spyOn(console, consoleProperty as 'Console').mockImplementation();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        jest.spyOn(serviceConnection as any, 'RequestHandshake').mockImplementation();
+
+        return testFn;
+    };
+
+    const createInputAction = (flag?: BitmaskFlags, position?: { x: number; y: number }) => {
+        const newFlag = flag ?? BitmaskFlags.MOVE;
+        const newPos = position ?? { x: 0, y: 0 };
+
+        return new WebsocketInputAction(
+            Date.now(),
+            BitmaskFlags.LEFT + BitmaskFlags.PRIMARY + newFlag + BitmaskFlags.PUSH,
+            newPos,
+            0,
+            0
+        );
+    };
+
+    beforeEach(() => {
+        // Reset service after each test to completely reset mocks
+        TouchFree.Init();
+        serviceConnection = ConnectionManager.serviceConnection();
+        if (serviceConnection) {
+            serviceConnection.webSocket.send = jest.fn((msg) => (message = msg as string));
+        }
+        message = '';
+        jest.restoreAllMocks();
+    });
+
+    it('should correctly handle a handshake warning', async () => {
+        const testFn = mockHandshake('warn');
+        mockOpen();
+
+        onMessage(ActionCode.VERSION_HANDSHAKE_RESPONSE, { message: 'Handshake Warning' });
+
+        await intervalTest(() => {
+            expect(testFn).toBeCalledWith('Received Handshake Warning from TouchFree:\n' + 'Handshake Warning');
+        });
+    });
+
+    it('should correctly handle a handshake error', async () => {
+        const testFn = mockHandshake('error');
+        mockOpen();
+
+        onMessage(ActionCode.VERSION_HANDSHAKE_RESPONSE, { message: 'Handshake Error', status: 'Error' });
+
+        await intervalTest(() => {
+            expect(testFn).toBeCalledWith('Received Handshake Error from TouchFree:\n' + 'Handshake Error');
+        });
+    });
+
+    it('should correctly check for a handshake response without a callback', async () => {
+        const consoleTestFn = jest.spyOn(console, 'warn').mockImplementation();
+        const consoleMessage = message;
+        const content = onMessage(ActionCode.VERSION_HANDSHAKE_RESPONSE);
+
+        await intervalTest(() => {
+            expect(consoleTestFn).toBeCalledWith(
+                'Received a Handshake Response that did not match a callback.' +
```

Comment:
I think testing the full content of this warning may make the tests a bit brittle. Though I don't think we will change them much in future

Reply:
Yeah I agree. Would testing just for the line "Received a Handshake Response that did not match a callback" be better?